### ISSUE TITLE
Incorrect Generated code for Inline Assembly

### ISSUE THREAD

**Issue by Keith-Cancel** [rust-lang/rust#156168]
The following code generates incorrect output. I would have expected the `-2` to generate an error, much like `mov {a}, [{a} - {b}]` would.  Instead the compiled output of the code below is ` mov {a}, [{a} + 2 * {b}]` which is **incorrect**! The minus `-` is switched to a `+`.  Also if someone is not paying attention this is extremely easy to miss since the code compiles and even runs. 

```rust
use core::arch::asm;
#[unsafe(no_mangle)]
pub fn example(a: usize, b: usize) {
    unsafe {
        asm!(
            "mov {a}, [{a} - 2 * {b}]",
            a  = in(reg) a,
            b  = in(reg) b,
            options(nostack),
        )  
    };
}
```
You can see the example out put here on Compiler explorer:
https://godbolt.org/z/zT44bbdc5

@rustbot label +T-compiler +A-inline-assembly

**Comment by asquared31415** [comment]
well that's a lovely llvm bug: <https://llvm.godbolt.org/z/G1TjoYfod>

@rustbot label +A-llvm

**Comment by Urgau** [comment]
I don't think we can do much in `rustc` about this miscompilation.

Opening an upstream [llvm/llvm-project](https://github.com/llvm/llvm-project) issue is probably the best way forward.

**Comment by el-ev** [comment]
`[{a} - 2 * {b}]` is not a valid x86 addressing mode. The scale must be 1, 2, 4, or 8, and cannot be negative. So this is an accept-invalid bug.

Upstream issue https://github.com/llvm/llvm-project/issues/96427 and pr https://github.com/llvm/llvm-project/pull/190460.
